### PR TITLE
Collect CUDA/CPU profiling info into result sheets.

### PR DESCRIPTION
**Pull request by golechwierowicz**
This PR:

0. Adds CUDA/CPU collection capabilties to the script.
1. Modifies result_analyzer.py to analyze newly collected results.
2. Moves CUDA synchronize/XLA device synchronize into the profiler.
3. Fixes list typing for Python 3.8+.

Tested with command:
```
python3 xla/benchmarks/experiment_runner.py --dynamo=openxla --xla=PJRT --test=train --filter=basic_gnn_gcn$ --suite-name=torchbench --accelerator=cuda --progress-bar --output-dirname=/tmp/output --repeat=2 --print-subprocess --no-resume --profile-cuda-cpu-collect --profile-cuda python3 xla/benchmarks/result_analyzer.py --output-dir=/tmp/output
```

### CODE REVIEW

File: benchmarks/experiment_runner.py [hunk 1]
```diff
@@ -249,8 +251,37 @@ def dump_profile_info(self, prof, model_name):
...
+        if evt.is_legacy:
+          if not use_device:
+            total_cuda_time += evt.self_cuda_time_total
+      elif evt.device_type == DeviceType.CUDA:
+        # in kineto profiler, there're events with the correct device type (e.g. CUDA)
+        total_cuda_time += evt.self_cuda_time_total
+
+    total_cpu_time /= 1000000
+    total_cuda_time /= 1000000
+    metrics["total_cpu_time"] = total_cpu_time
```
Comment (frgossen) [inline benchmarks/experiment_runner.py:279]: Can we add a suffix indicating the unit? 
us in this case?

**Review by frgossen — COMMENTED** [review]
Nice, ty!

**Review by frgossen — APPROVED** [review]
LGTM, one comment.

File: benchmarks/experiment_runner.py [hunk 2]
```diff
@@ -249,8 +251,41 @@ def dump_profile_info(self, prof, model_name):
...
-  def timed_run(self, benchmark_experiment, benchmark_model):
+  def collect_profile_to_metrics(self, prof, metrics):
+    assert prof is not None, 'Expecting profiler to be defined!'
+    if not self._args.profile_cuda_cpu_collect:
+      logger.warning(
+          'Profiling enabled, but collection of CPU/CUDA profiling info disabled.'
+      )
+      return
+
+    kernel_dump = prof.profiler.total_average()
```
Comment (vanbasten23) [inline benchmarks/experiment_runner.py:262]: Curious where is this total_average() defined?

Comment (golechwierowicz) [reply benchmarks/experiment_runner.py:262]: https://github.com/pytorch/pytorch/blob/00412e6dfacdec5a8508c841b3ea0846388dc872/torch/autograd/profiler_util.py#L339-L350

**Comment by miladm** [comment]
Thanks.

cc @zpcore to take advantage of this feature in future benchmarking automation work.
